tests/it/cluster/client_test.go

```diff
@@ -642,3 +656,80 @@ func TestClientPeekLogStream(t *testing.T) {
 	_, _, err = client.PeekLogStream(context.Background(), tpid, lsid)
 	require.Error(t, err)
 }
+
+func TestClientAppendWithAllowedLogStream(t *testing.T) {
+	const numLogs = 100
+	const numLogStreams = 10
+
+	clus := it.NewVarlogCluster(t,
+		it.WithNumberOfStorageNodes(1),
+		it.WithNumberOfLogStreams(numLogStreams),
+		it.WithNumberOfClients(1),
+		it.WithVMSOptions(it.NewTestVMSOptions()...),
+		it.WithNumberOfTopics(1),
+	)
+
+	defer func() {
+		clus.Close(t)
+		testutil.GC()
+	}()
+
+	topicID := clus.TopicIDs()[0]
+	logStreamID := clus.LogStreamIDs(topicID)[0]
+	sealedLogStreamID := clus.LogStreamIDs(topicID)[1]
+	client := clus.ClientAtIndex(t, 0)
+
+	allowedLogStreams := make(map[types.LogStreamID]struct{})
+	allowedLogStreams[logStreamID] = struct{}{}
+	allowedLogStreams[sealedLogStreamID] = struct{}{}
+
+	rsp, err := clus.Seal(t, topicID, sealedLogStreamID)
+	require.NoError(t, err)
+	require.Len(t, rsp.LogStreams, 1)
+	require.Equal(t, varlogpb.LogStreamStatusSealed, rsp.LogStreams[0].Status)
+
+	for i := 0; i < numLogs; i++ {
+		res := client.Append(context.Background(), topicID, [][]byte{[]byte("foo")}, varlog.WithAllowedLogStreams(allowedLogStreams))
+		require.NoError(t, res.Err)
+	}
+
+	// SubscribeTo [2, 1)
+	subscriber := client.SubscribeTo(context.Background(), topicID, logStreamID, types.LLSN(2), types.LLSN(1))
+	_, err = subscriber.Next()
+	require.Error(t, err)
+	require.NoError(t, subscriber.Close())
+
+	// SubscribeTo [1, 101)
+	subscriber = client.SubscribeTo(context.Background(), topicID, logStreamID, types.MinLLSN, types.LLSN(numLogs+1))
+	for i := 0; i < numLogs; i++ {
+		logEntry, err := subscriber.Next()
+		require.NoError(t, err)
+		require.Equal(t, topicID, logEntry.TopicID)
+		require.Equal(t, logStreamID, logEntry.LogStreamID)
+		require.Equal(t, types.LLSN(i+1), logEntry.LLSN)
+	}
+	_, err = subscriber.Next()
+	require.ErrorIs(t, err, io.EOF)
+	require.NoError(t, subscriber.Close())
+
+	// SubscribeTo [1, max)
+	subscriber = client.SubscribeTo(context.Background(), topicID, logStreamID, types.MinLLSN, types.MaxLLSN)
+	for i := 0; i < numLogs; i++ {
+		logEntry, err := subscriber.Next()
+		require.NoError(t, err)
+		require.Equal(t, topicID, logEntry.TopicID)
+		require.Equal(t, logStreamID, logEntry.LogStreamID)
+		require.Equal(t, types.LLSN(i+1), logEntry.LLSN)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		_, err := subscriber.Next()
+		require.Error(t, err)
+		require.NotErrorIs(t, err, io.EOF)
+	}()
+	time.Sleep(5 * time.Millisecond)
+	require.NoError(t, subscriber.Close())
```

Comment:
Checking subscribing logs with extensive ranges seems out of the test scope.

Reply:
removed